sdks/java/io/pulsar/src/main/java/org/apache/beam/sdk/io/pulsar/PulsarMessage.java

```diff
@@ -0,0 +1,37 @@
+package org.apache.beam.sdk.io.pulsar;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.pulsar.client.api.Message;
+
+/**
+ * Class representing a Pulsar Message record. Each PulsarMessage contains a single message basic message data
+ * and Message record to access directly.
+ */
+public class PulsarMessage {
+    private String topic;
+    private Long publishTimestamp;
+    private Object messageRecord;
```

Comment:
it is not used anywhere

Reply:
The purpose of PulsarMessage is to handle a [coder](https://beam.apache.org/documentation/programming-guide/#specifying-coders) for the output in the beam pipeline (see PulsarMessageCoder). So at the beginning I was delivering the Message as the output of the Pipeline, but you need to specify a coder for every pipeline in beam, so I generate the wrapper of the Message so you can set a custom coder for the message.